src/modules/adaptors/Morpho/MorphoBlue/MorphoBlueCollateralAdaptor.sol

```diff
@@ -0,0 +1,230 @@
+ *      Blue
+ * @author crispymangoes, 0xEinCodes
+ * TODO - The periphery libraries from MB may be used depending on how much gas they use. For now we will not use them but we will test to see which is more gas efficient.
+ */
+contract MorphoBlueCollateralAdaptor is BaseAdaptor, MorphoBlueHealthFactorLogic {
+    using SafeTransferLib for ERC20;
+    using Math for uint256;
+
+    //==================== Adaptor Data Specification ====================
+    // adaptorData = abi.encode(Id id)
+    // Where:
+    // `id` is the var defined by Morpho Blue for the bytes identifier of a Morpho Blue market
+    //================= Configuration Data Specification =================
+    // NA
+    //====================================================================
+
+    /**
+     * @notice Attempted to interact with an Morpho Blue Lending Market the Cellar is not using.
+     */
+    error MorphoBlueCollateralAdaptor__MarketPositionsMustBeTracked(Id id);
+
+    /**
+     * @notice Removal of collateral causes Cellar Health Factor below what is required
+     */
+    error MorphoBlueCollateralAdaptor__HealthFactorTooLow(Id id);
+
+    /**
+     * @notice Minimum Health Factor enforced after every removeCollateral() strategist function call.
+     * @notice Overwrites strategist set minimums if they are lower.
+     */
+    uint256 public immutable minimumHealthFactor;
+
+    constructor(address _morphoBlue, uint256 _healthFactor) MorphoBlueHealthFactorLogic(_morphoBlue) {
+        _verifyConstructorMinimumHealthFactor(_healthFactor);
+        morphoBlue = IMorpho(_morphoBlue);
+        minimumHealthFactor = _healthFactor;
+    }
+
+    //============================================ Global Functions ===========================================
+    /**
+     * @dev Identifier unique to this adaptor for a shared registry.
+     * Normally the identifier would just be the address of this contract, but this
+     * Identifier is needed during Cellar Delegate Call Operations, so getting the address
+     * of the adaptor is more difficult.
+     */
+    function identifier() public pure virtual override returns (bytes32) {
+        return keccak256(abi.encode("Morpho Blue Collateral Adaptor V 0.1"));
+    }
+
+    //============================================ Implement Base Functions ===========================================
+    /**
+     * @notice User deposits collateralToken to Morpho Blue market
+     * @param assets the amount of assets to provide as collateral on Morpho Blue
+     * @param adaptorData adaptor data containing the abi encoded Id for specific Morpho Blue Lending Market
+     * @dev configurationData is NOT used
+     */
+    function deposit(uint256 assets, bytes memory adaptorData, bytes memory) public override {
+        // Deposit assets to Morpho Blue.
+        Id id = abi.decode(adaptorData, (Id));
+        _validateMBMarket(id);
+
+        MarketParams memory market = morphoBlue.idToMarketParams(id);
```

Comment:
How much state does this read? I wonder if it would be better to just have the adaptor data store the ID and the loan token

Reply:
I'll have to look into this, but that might just be an easier thing to do and could reduce gas.

Reply:
This call to `idToMarketParams` should cost 5 cold SLOADs, so 10k alone (the storage slots loaded here are not used by Blue anywhere). Ideally yes, the required data is encoded in `adaptorData`, so it costs much less (this is the design we've chosen for Blue)

Reply:
It is also possible to have the adaptor data store the MarketParams instead. The call to `MarketParamsLib.id(marketParams)` is way cheaper than the call to `idToMarketParams(id)` as it doesn't read storage.

Reply:
Thank for you pointing this out, we are looking into it :)